Game stopped working after last update (Android 9)

Hi guys,
First thanks for the great game, and very good recent improvements!

After one of the most recent updates, it totally does not start up. Just displays the application crashed.
I'm installing from F-Droid, and Motorola Moto X4 phone, running Android 9 (Amazon).
Thanks in advance!

Hi,
This is output "logcat -b crash" based on the link you sent. This appears when I attempt to start the game. The phone is set to Hungarian language, there is no translation yet.

06-01 20:47:22.261 17059 17059 E AndroidRuntime: FATAL EXCEPTION: main
06-01 20:47:22.261 17059 17059 E AndroidRuntime: Process: com.serwylo.beatgame, PID: 17059
06-01 20:47:22.261 17059 17059 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.serwylo.beatgame/com.serwylo.beatgame.AndroidLauncher}: java.lang.NullPointerException: Attempt to invoke interface method 'void com.badlogic.gdx.Application.error(java.lang.String, java.lang.String)' on a null object reference
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3121)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3260)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1976)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:193)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6912)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:860)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void com.badlogic.gdx.Application.error(java.lang.String, java.lang.String)' on a null object reference
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at com.serwylo.beatgame.Assets$Companion.getLocale(Assets.kt:314)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at com.serwylo.beatgame.BeatFeetGame.<init>(BeatFeetGame.kt:15)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at com.serwylo.beatgame.AndroidLauncher.onCreate(AndroidLauncher.java:13)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7148)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7139)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1293)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3101)
06-01 20:47:22.261 17059 17059 E AndroidRuntime: 	... 11 more

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -20,8 +20,8 @@ android {
         applicationId "com.serwylo.beatgame"
         minSdkVersion 14
         targetSdkVersion 30
-        versionCode 10 
-        versionName "0.10.0"
+        versionCode 11 
+        versionName "0.10.1"
     }
     buildTypes {
         release {

--- a/core/src/com/serwylo/beatgame/BeatFeetGame.kt
+++ b/core/src/com/serwylo/beatgame/BeatFeetGame.kt
@@ -12,13 +12,18 @@ import java.util.*
 
 open class BeatFeetGame(val platformListener: PlatformListener, private val verbose: Boolean) : Game() {
 
-    val assets = Assets(Assets.getLocale())
+    // Initialize this in the create() method so that we can access Gdx logging. Helps to diagnose
+    // issues with asset loading if we can log meaningful messages.
+    // See https://github.com/beat-feet/beat-feet/issues/97.
+    lateinit var assets: Assets
 
     @Suppress("LibGDXLogLevel") // Optional flag to make more verbose.
     override fun create() {
         if (verbose) {
             Gdx.app.logLevel = Application.LOG_DEBUG
         }
+
+        assets = Assets(Assets.getLocale())
 
         Globals.shapeRenderer = ShapeRenderer()
         Globals.spriteBatch = SpriteBatch()

--- a/fastlane/metadata/android/en-US/changelogs/11.txt
+++ b/fastlane/metadata/android/en-US/changelogs/11.txt
@@ -1,0 +1,5 @@
+Fixed critical crash when Android system settings are set to a language unsupported by Beat Feet (thanks to lionkmp for reporting and helping to investigate).
+
+Updated translations.
+
+Support via GitHub Sponsors or Liberapay.

